src/tasks/base.py

```diff
@@ -101,12 +102,15 @@ def __check_drop_last_params(self, data_params: List[Dict[str, Any]], phase: str
                 raise ValueError(f'DataLoader parametrs `drop_last` must be False in {phase} phase.')
 
     def on_train_start(self) -> None:
-        # TODO check and load checkpoint
-        pass
+        if self.current_epoch == 0:
+            load_checkpoint(self, base_ckpt_path=self.hparams.task.base_checkpoint, 
+                            override_name2ckpt_path=self.hparams.task.override_checkpoints,
```

Comment:
hparams is protected self._hparams.task.base_checkpoint e.t.c.. It's not good call property in class